Sources/muterCore/CLICommands/RunCommand/RunCommand.swift

```diff
@@ -1,84 +1,65 @@
-import Commandant
-import Result
-import Curry
 import Foundation
+import ArgumentParser
 
-@available(OSX 10.13, *)
-public struct RunCommand: CommandProtocol {
+@available(macOS 10.13, *)
+public struct Run: ParsableCommand {
+    public static let configuration = CommandConfiguration(
+        commandName: "run",
+        abstract: "Performs mutation testing for the Swift project contained within the current directory"
+    )
 
-    public typealias Options = RunCommandOptions
-    public typealias ClientError = MuterError
-    public let verb: String = "run"
-    public let function: String = """
-    Performs mutation testing for the Swift project contained within the current directory
+    @Option(help: "Only mutate a given list of source code files")
+    var filesToMutate: [String] = []
 
-    Muter defaults to run when you don't specify any subcommands
+    @Flag(name: [.customLong("output-json")], help: "Output test results to a json file.")
+    var shouldOutputJson: Bool = false
 
-    Available flags:
+    @Flag(name: [.customLong("output-xcode")], help: "Output test results in a format consumable by an Xcode run script step.")
+    var shouldOutputXcode: Bool = false
 
-       --files-to-mutate    Only mutate a given list of source code files
-       --output-json        Output test results to a json file
-       --output-xcode       Output test results in a format consumable by an Xcode run script step
+    public init() { }
 
-    """
-    
-    private let fileManager: FileSystemManager
-    private let notificationCenter: NotificationCenter
+    public func run() throws {
+        let _ = RunCommandObserver(
+            reporter: Reporter(
+                shouldOutputJson: shouldOutputJson,
+                shouldOutputXcode: shouldOutputXcode
+            ),
+            fileManager: FileManager.default,
+            flushHandler: flushStdOut
+        )
 
-    public init(fileManager: FileSystemManager = FileManager.default,
-                notificationCenter: NotificationCenter = .default) {
-        self.fileManager = fileManager
-        self.notificationCenter = notificationCenter
-    }
-
-    public func run(_ options: Options) -> Result<(), ClientError> {
-        let _ = RunCommandObserver(reporter: options.reporter,
-                                   fileManager: fileManager,
-                                   flushHandler: flushStdOut)
+        NotificationCenter.default.post(name: .muterLaunched, object: nil)
         
-        notificationCenter.post(name: .muterLaunched, object: nil)
-
-        let result = RunCommandHandler(options: options).handle()
-        switch result {
-        case .success(_):
-            return .success(())
-        case .failure(let error):
-            return .failure(error)
+        do {
+            try RunCommandHandler(command: self).handle()
+        } catch {
+            
+            throw CleanExit.message(
```

Comment:
We should probably exit with the error code from `error` rather than a clean exit. Maybe `MuterError` could return error codes, or just use a single generic code like -1.

Reply:
I've opt for clean exit because anything else would prepend `Error: ` with the error message